Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp

```diff
@@ -39,15 +39,15 @@ namespace AzQtComponents
         return AZ::Color(static_cast<float>(rgb.redF()), static_cast<float>(rgb.greenF()), static_cast<float>(rgb.blueF()), static_cast<float>(rgb.alphaF()));
     }
 
-    QString toString(double value, int numDecimals, const QLocale& locale, bool showGroupSeparator)
+    QString toString(double value, int numDecimals, const QLocale& locale, bool showGroupSeparator, bool truncate)
     {
         const QChar decimalPoint = locale.decimalPoint();
         const QChar zeroDigit = locale.zeroDigit();
         const int numToStringDecimals = AZStd::max(numDecimals, 20);
 
-        // We want to truncate, not round. toString will round, so we add extra decimal places to the formatting
-        // so we can remove the last values
-        QString retValue = locale.toString(value, 'f', (numDecimals > 0) ? numToStringDecimals : 0);
+        // If we want to truncate, not round, we add extra decimal places to the formatting
+        // so we can remove the last values otherwise we allow rounding
+        QString retValue = locale.toString(value, 'f', (numDecimals > 0) ? (truncate ? numToStringDecimals : numDecimals) : 0);
```

Comment:
suggestion: Could we replace the nested ternary statement here with an if statement (or immediately invoked lambda) It just gets pretty hard to parse as it is right now

Reply:
Done.

Reply:
Legend, thanks 🙂